src/Microsoft.AspNetCore.Authentication.Cookies/CookieAuthenticationHandler.cs

```diff
@@ -29,10 +29,17 @@ public class CookieAuthenticationHandler : SignInAuthenticationHandler<CookieAut
         private string _sessionKey;
         private Task<AuthenticateResult> _readCookieTask;
         private AuthenticationTicket _refreshTicket;
+        private readonly ITicketStore _diSessionStore;
 
-        public CookieAuthenticationHandler(IOptionsMonitor<CookieAuthenticationOptions> options, ILoggerFactory logger, UrlEncoder encoder, ISystemClock clock)
+        private ITicketStore GetSessionStore() => Options.UseSessionStoreFromDI
+            ? _diSessionStore ?? Options.SessionStore
+            : Options.SessionStore;
+
+        public CookieAuthenticationHandler(IOptionsMonitor<CookieAuthenticationOptions> options, ILoggerFactory logger, UrlEncoder encoder, ISystemClock clock, ITicketStore diSessionStore = default)
```

Comment:
No API signature changes are allowed in 2.1. You'll have to add a new constructor.

Reply:
@pakrym does DI support optional parameters like this? We don't usually use them. We tend to use `IEnumerable<T>` for optional services.

Reply:
It does, usually, we use multiple constructors to achieve optional args. Usage of IEnumerable optional services is generally discouraged.